pkg/clusteragent/autoscaling/workload/controller_horizontal.go

```diff
@@ -222,6 +222,15 @@ func (hr *horizontalController) computeScaleAction(
 	// Now that we have applied all modifications to targetDesiredReplicas, we can compute the scale direction
 	scaleDirection := common.GetScaleDirection(currentDesiredReplicas, targetDesiredReplicas)
 
+	// Check if we are in fallback mode and scaling direction is disabled
```

Comment:
nit: Technically you only need to check that if `scaleDirection != NoScale`, as otherwise there's nothing to do and scaling is not actually disabled.